applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h

```diff
@@ -155,10 +152,10 @@ class DampingUtilities
         KRATOS_INFO("ShapeOpt") << "Starting to prepare damping..." << std::endl;
 
         // Loop over all regions for which damping is to be applied
-        for (unsigned int regionNumber = 0; regionNumber < len(mrDampingRegions); regionNumber++)
+        for (unsigned int regionNumber = 0; regionNumber < mDampingSettings["damping_regions"].size(); regionNumber++)
```

Comment:
```suggestion
        for (const auto& r_region_parameters : mDampingSettings["damping_regions"])
```

This works as well, and cleaner IMO. :)

Reply:
much better

Reply:
@sunethwarna I am pretty sure this doesn’t work
Remember we discussed this not too long ago?
I can dig up the issue, @armingeiser in the meantime I would check carefully, test it with multiple entries in this array

Reply:
good to know, thanks for making me aware. I will check

Reply:
Hm, it works for 1-6 entries i did not test for more.

Reply:
as far as i can see the parameters object implements all required things (begin, end, ++, !=, *)

Reply:
I meant this one: #1466
Does it work now or does the issue describe sth different?

Reply:
The loop here is in cpp, so the discussion from #1466 does describe something different since its related to python.

Nevertheless i have also tried to loop over it in python (`for i in ...`) and it also worked in my case.

Reply:
so you mean the issue from #1466 no longer persists?
That would be cool

Reply:
I checked, yes, it works in both c++ and python level now. I am wondering is, why it did not work earlier, I checked the changes, these iterators were implemented 3 eyars ago.